### PR TITLE
check_systemd: init at 2.2.1

### PR DESCRIPTION
**Pull request by symphorien**
and remove pynagsystemd, which it replaces.

pynagsystemd is not compatible with nixos's version of systemd, and is unmaintained. check_systemd is a maintained fork of it, which is compatible with systemd 246.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -463,6 +463,7 @@ mapAliases ({
   phonon-backend-gstreamer = throw "phonon-backend-gstreamer: Please use libsForQt5.phonon-backend-gstreamer, as Qt4 support in this package has been removed."; # added 2019-11-22
   phonon-backend-vlc = throw "phonon-backend-vlc: Please use libsForQt5.phonon-backend-vlc, as Qt4 support in this package has been removed."; # added 2019-11-22
   phonon = throw "phonon: Please use libsForQt5.phonon, as Qt4 support in this package has been removed."; # added 2019-11-22
+  pynagsystemd = throw "pynagsystemd was removed as it was unmaintained and incompatible with recent systemd versions. Use check_systemd, its fork, instead."; # added 2020-10-24
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:466]:
```suggestion
  pynagsystemd = throw "pynagsystemd was removed as it was unmaintained and incompatible with recent systemd versions. Instead use its fork check_systemd."; # added 2020-10-24
```

**Comment by SuperSandro2000** [comment]
Can you fix the eval error?

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102271` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>check_systemd</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102271` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>check_systemd</li>
  </ul>
</details>

**Comment by symphorien** [comment]
I restricted the platform to linux, as systemd only supports linux anyway.

File: pkgs/servers/monitoring/nagios/plugins/check_systemd.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ fetchFromGitHub, python3Packages, lib }:
+let
+  version = "2.2.1";
+in
+python3Packages.buildPythonApplication rec {
+  pname = "check_systemd";
+  inherit version;
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/nagios/plugins/check_systemd.nix:7]:
```suggestion

python3Packages.buildPythonApplication rec {
  pname = "check_systemd";
  version = "2.2.1";
```

File: pkgs/servers/monitoring/nagios/plugins/check_systemd.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    rev = "v${version}";
+    sha256 = "04r14dhqzrdndn235dvr6afy4s4g4asynsgvj99cmyq55nah4asn";
+  };
+
+  propagatedBuildInputs = with python3Packages; [ nagiosplugin ];
+
+  postInstall = ''
+    # check_systemd is only a broken stub calling check_systemd.py
+    rm $out/bin/check_systemd
+    mv $out/bin/check_systemd{.py,}
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/nagios/plugins/check_systemd.nix:21]:
```suggestion
    mv $out/bin/check_systemd{.py,}
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102271` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>check_systemd</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102271` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>check_systemd</li>
  </ul>
</details>

**Pull request merged**